### PR TITLE
Reflection changes, service interfaces, configured test directory

### PR DESCRIPTION
**Pull request by xXxF0RR35TC4MPxXx**
Added xUnit and MOQ to solution, created an example test and it passes correctly. We want to push it to main so other group members can work on their own controller tests.

### CODE REVIEW

File: Tests/AuthTests/Class1.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Tests.AuthTests
+{
+    internal class Class1
```
Comment (tkarelus) [inline Tests/AuthTests/Class1.cs:9]: You need this file?

Comment (xXxF0RR35TC4MPxXx) [reply Tests/AuthTests/Class1.cs:9]: It's there only for the folders to be visible on github. That's the only reason. Same with other Class1.cs files in other folders.

File: Tests/CandidateTests/GetTests.cs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+                _logger.Object,
+                _mapper.Object
+                );
+
+            var dto = new CandidateProfileDTO(1, "fullName", "email", "phone", DateTime.Now, 1, "", "");
+
+            con.Get(dto.Id);
+
+            _candidateService.Verify(g => g.GetCandidateProfileById(dto.Id), Times.Once);
+
```
Comment (tkarelus) [inline Tests/CandidateTests/GetTests.cs]: You don't any empty line here

File: Tests/CandidateTests/GetTests.cs [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+
+        }
+
```
Comment (tkarelus) [inline Tests/CandidateTests/GetTests.cs]: You don't any empty line here

**Review by tkarelus — APPROVED** [review]

**Pull request merged**